src/hippopt/robot_planning/utilities/interpolators.py

```diff
@@ -0,0 +1,456 @@
+            f"Initial state has {len(initial_state.positions)} joints, "
+            f"but final state has {len(final_state.positions)} joints."
+        )
+
+    positions_interpolation = linear_interpolator(
+        initial=initial_state.positions,
+        final=final_state.positions,
+        number_of_points=number_of_points,
+    )
+    output = []
+    for i in range(number_of_points):
+        output_state = KinematicTreeState(
+            number_of_joints_state=len(initial_state.positions)
+        )
+        output_state.positions = positions_interpolation[i]
+        output.append(output_state)
+    return output
+
+
+def floating_base_system_state_interpolator(
+    initial_state: FloatingBaseSystemState,
+    final_state: FloatingBaseSystemState,
+    number_of_points: int,
+) -> list[FloatingBaseSystemState]:
+    base_interpolation = free_floating_object_state_interpolator(
+        initial_state=initial_state.base,
+        final_state=final_state.base,
+        number_of_points=number_of_points,
+    )
+    joints_interpolation = kinematic_tree_state_interpolator(
+        initial_state=initial_state.joints,
+        final_state=final_state.joints,
+        number_of_points=number_of_points,
+    )
+
+    assert len(base_interpolation) == len(joints_interpolation) == number_of_points
+
+    output = []
+    for i in range(number_of_points):
+        output_state = FloatingBaseSystemState(
+            number_of_joints_state=len(initial_state.joints.positions)
+        )
+        output_state.base = base_interpolation[i]
+        output_state.joints = joints_interpolation[i]
+        output.append(output_state)
+    return output
+
+
+def humanoid_state_interpolator(
+    initial_state: HumanoidState,
+    final_state: HumanoidState,
+    contact_phases: FeetContactPhasesDescriptor,
+    contact_descriptor: FeetContactPointDescriptors,
+    number_of_points: int,
+    dt: float,
+    t0: float = 0.0,
+):
+    contact_points_interpolation = feet_contact_points_interpolator(
+        phases=contact_phases,
+        descriptor=contact_descriptor,
+        number_of_points=number_of_points,
+        dt=dt,
+        t0=t0,
+    )
+    kinematics_interpolation = floating_base_system_state_interpolator(
+        initial_state=initial_state.kinematics,
+        final_state=final_state.kinematics,
+        number_of_points=number_of_points,
+    )
+    com_interpolation = linear_interpolator(
+        initial=initial_state.com,
+        final=final_state.com,
+        number_of_points=number_of_points,
+    )
+
+    assert (
+        len(contact_points_interpolation)
+        == len(kinematics_interpolation)
+        == len(com_interpolation)
+        == number_of_points
+    )
+
+    output = []
+    for i in range(number_of_points):
+        output_state = HumanoidState(
+            contact_point_descriptors=contact_descriptor,
+            number_of_joints=len(initial_state.kinematics.joints.positions),
+        )
```

Comment:
usual comment on pattern:

```suggestion
    output_state = [
        HumanoidState(
            contact_point_descriptors=contact_descriptor,
            number_of_joints=len(initial_state.kinematics.joints.positions),
        )
        for _ in range(number_of_points)
     ]
```

Reply:
Thanks, I have done it in https://github.com/ami-iit/hippopt/pull/10/commits/12480c5acdb7d51b35e0d7899390ed0f143e90c2